File: done/find-the-divisors.py
# https://www.codewars.com/kata/find-the-divisors/train/python
from unittest import TestCase


def divisors(integer):
    limit = int(integer * 0.5) + 1
    is_divisor = lambda x: integer % x == 0
    return list(filter(is_divisor, range(2, limit))) or "{} is prime".format(integer)


test = TestCase()
test.assertEqual(divisors(15), [3, 5]);
test.assertEqual(divisors(12), [2, 3, 4, 6]);
test.assertEqual(divisors(13), "13 is prime");

File: done/kyu-7/find-the-divisors.ex
# https://www.codewars.com/kata/find-the-divisors/train/elixir

defmodule FindTheDivisors do
  def divisors(2), do: [2]

  def divisors(integer) do
    limit = (integer / 2) |> Kernel.trunc() |> Kernel.+(1)

    divisors =
      2..limit
      |> Enum.to_list()
      |> Enum.filter(fn x -> rem(integer, x) == 0 end)

    if Enum.empty?(divisors) do
      "#{integer} is prime"
    else
      divisors
    end
  end
end

IO.inspect(FindTheDivisors.divisors(12))
